sub-packages/bionemo-esm2/src/bionemo/esm2/model/finetune/loss.py

```diff
@@ -0,0 +1,132 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Dict, Sequence, Tuple
+
+import torch
+from megatron.core import parallel_state
+from torch import Tensor
+
+from bionemo.llm.model.loss import BERTMLMLossWithReduction, PerTokenLossDict, SameSizeLossDict
+
+
+__all__: Sequence[str] = (
+    "RegressorLossReduction",
+    "ClassifierLossReduction",
+)
+
+
+class RegressorLossReduction(BERTMLMLossWithReduction):
+    """A class for calculating the MSE loss of regression output.
+
+    This class used for calculating the loss, and for logging the reduced loss across micro batches.
+    """
+
+    def forward(
+        self, batch: Dict[str, Tensor], forward_out: Dict[str, Tensor]
+    ) -> Tuple[Tensor, PerTokenLossDict | SameSizeLossDict]:
+        """Calculates the loss within a micro-batch. A micro-batch is a batch of data on a single GPU.
+
+        Args:
+            batch: A batch of data that gets passed to the original forward inside LitAutoEncoder.
+            forward_out: the output of the forward method inside classification head.
+
+        Returns:
+            A tuple containing [<loss_tensor>, ReductionT] where the loss tensor will be used for
+                backpropagation and the ReductionT will be passed to the reduce method
+                (which currently only works for logging.).
+        """
+        regression_output = forward_out["regression_output"]
+        targets = batch["labels"].to(dtype=regression_output.dtype)  # [b, 1]
+
+        cp_size = parallel_state.get_context_parallel_world_size()
+        if cp_size == 1:
+            loss = torch.nn.functional.mse_loss(regression_output, targets)
+        else:  # TODO: support CP with masked_token_loss_context_parallel
```

Comment:
If you're going to add a TODO comment we should link a Jira ticket or github issue

Reply:
hm, this was already there, I think we can remove the comment since we don't expect context parallel here